src/rt/mod.rs

```diff
@@ -73,8 +73,27 @@ where
     trace!(thread = ?id, "spawn");
 
     Scheduler::spawn(Box::new(move || {
+        /// the given closure `f` may panic when executed.
+        /// when this happens, we still want to ensure that
+        /// thread locals are destructed. therefore, we set
+        /// up a guard that is dropped as part of the unwind
+        /// logic when `f` panics.
```

Comment:
These should be regular comments, not doc comments.

Reply:
These doc comments intentionally apply to the `struct PanicGuard;`